cloudbridge/cloud/providers/gce/resources.py

```diff
@@ -1639,12 +1640,22 @@ def name(self):
         """
         return self._volume.get('name')
 
+    @name.setter
+    # pylint:disable=arguments-differ
+    def name(self, value):
+        GCEVolume.assert_valid_resource_name(value)
+        # In GCE, the name of the volume is provided by the client when
+        # initially creating the resource. The name cannot be changed after
+        # the volume is created.
+        cb.log.warning("Setting volume name after it is created is not "
```

Comment:
@baizhang Can we use a label/tag to track the name instead? In cloudbridge, the name is just a user provided friendly name, and is therefore, changeable. So we could do something similar to what's done for description below? (unless it's super inefficient, and requires a new http request)

Reply:
Hi @nuwang -

In https://github.com/gvlproject/cloudbridge/pull/44, where we landed was:

> @nuwang - your proposal seems reasonable to me:
> 
> Use the disk name field for name (with the specified character and inability to rename restrictions)
> Use labels for the disk description field. GCE implementation will convert spaces to underscores
 
Please review the discussion. Do the premises of the discussion in that PR still hold or do we need to revisit?

Thanks,

-Matt

Reply:
@mbookman Thanks for pointing to that. Going through that discussion through, I can't find why exactly we decided to go with the GCE name for the 'name' field, instead of using a label, but I think there was some important reason? As I recall, @afgane raised an issue - something to do with potential mismatches between the original name given at creation time and a later rename perhaps? It would be good to record what it is for future reference.

Reply:
And - not all Cloud resource support labels. So we could potentially end up with a hodge podge of resources which did support renames vs. those that did not.

Reply:
@mbookman Sorry, I had missed this comment. Right, that was the issue - thanks for penning it down. I guess we still haven't figured out a way to make this consistent across all providers, since, as it stands now, a rename will fail silently for GCE only. Let's leave this as is for now, since it's not a major issue, and hopefully, figure out a way to make things uniform across all providers if and when needed?